### ISSUE TITLE
 java.lang.UnsatisfiedLinkError: org.rocksdb.TransactionDBOptions.newTransactionDBOptions()

### ISSUE THREAD

**Issue by zxgangandy** [facebook/rocksdb#3788]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

I use the java version as below:
```
<dependency>
            <groupId>org.rocksdb</groupId>
            <artifactId>rocksdbjni</artifactId>
            <version>5.11.3</version>
 </dependency>
```

On Windows 10.

### Actual behavior

code : 
```
final  TransactionDBOptions txnDbOptions = new TransactionDBOptions();
```

It will show the errors: 
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.TransactionDBOptions.newTransactionDBOptions()J
	at org.rocksdb.TransactionDBOptions.newTransactionDBOptions(Native Method)
	at org.rocksdb.TransactionDBOptions.<init>(TransactionDBOptions.java:11)
```

Does anyone can do some help?

**Comment by zxgangandy** [comment]
The system is windows 10.

**Comment by SivagurunathanV** [comment]
Any update on this? I am still facing this issue.

Thanks.

**Comment by lucassaldanha** [comment]
Same problem here. I've tried installing"Microsoft Visual C++ 2015 Redistributable" as per the Windows instructions but I still have the same problem.

**Environment**
RocksDB 5.14.2
Windows 10 x64

**Error**
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.TransactionDBOptions.newTransactionDBOptions()J
        at org.rocksdb.TransactionDBOptions.newTransactionDBOptions(Native Method)
        at org.rocksdb.TransactionDBOptions.<init>(TransactionDBOptions.java:11)
        at net.consensys.pantheon.services.kvstore.RocksDbKeyValueStorage.<init>(RocksDbKeyValueStorage.java:46)
        at net.consensys.pantheon.services.kvstore.RocksDbKeyValueStorage.create(RocksDbKeyValueStorage.java:40)
```

**Comment by lucassaldanha** [comment]
Ok, I think I know what is going on...

The last maven release (5.14.2) was on the 5th July 2018 (https://github.com/facebook/rocksdb/releases/tag/v5.14.2).

On 19th July 2018, there was a PR adding some missing libraries into the java/CMakeLists.txt file
(https://github.com/facebook/rocksdb/commit/80afa8490371ad50d65e10063fc576a7839e7628#diff-4bed60f270bfee21f80ac14ab5bf7f28)

One of the addition is the `rocksjni/transaction_db_options.cc`, that is exactly where I'm getting the error.

So the `librocksdbjni-win64.dll` distributed on the lates available Maven version doesn't contain the method I'm trying to execute, hence the `UnsatisfiedLinkError`.

@adamretter Do you know when a new Maven version will be released? I would love to keep using your maven artifacts.

**Comment by lucassaldanha** [comment]
Ok guys,

I was able to get this working building the latest version from source. I've followed the wiki instructions to [build on Windows](https://github.com/facebook/rocksdb/wiki/Building-on-Windows), however, I had to set the compression libraries flags to false. I also used the PORTABLE flag.

This is the cmake command that I used:
```
cmake -G "Visual Studio 14 Win64" -DJNI=1 -DGFLAGS=1 -DSNAPPY=0 -DLZ4=0 -DZLIB=0 -DXPRESS=0 -DPORTABLE=1 ..
```

After that, I added the shared DLL that was built into the RocksDB jar, overrinding the DDL that was there before.
